src/components/Cart/Cart.js

```diff
@@ -0,0 +1,198 @@
+import {
+  ArrowLeftOutlined,
+  ShoppingCartOutlined,
+  AlipayCircleOutlined,
+  CheckCircleOutlined,
+  DeleteOutlined,
+  PlusCircleOutlined,
+  MinusCircleOutlined,
+} from "@ant-design/icons";
+import { Steps, notification } from "antd";
+import "./Cart.css";
+import { useDispatch, useSelector } from "react-redux";
+import React, { useState, useEffect } from "react";
+import {
+  inccQuantity,
+  removeFromCart,
+  descQuantity,
+  cartTotal,
+} from "../../redux/actions/productActions";
+import { Link } from "react-router-dom";
+
+const { Step } = Steps;
+
+function Cart() {
+  // const cart = useSelector((state) => state.allProducts.cart);
+  const grandTotal = useSelector((state) => state.cartTotal);
+  const {cart}=useSelector((state)=>state.cart);
+
+  // console.log(grandTotal)
+  const dispatch = useDispatch();
+  const [total, setTotal] = useState(0);
+
+  useEffect(() => {
+    setTotal(
+      cart.reduce((acc, curr) => acc + Number(curr.price) * curr.qty, 0)
+    );
+    dispatch(cartTotal({ total: total }));
+  }, [cart, total]);
+
+  console.log(grandTotal.total);
+  console.log(cart);
+
+  const deleteItem = () => {
+    notification.error({
+      message: "Item Deleted",
+      placement: "top",
+      duration: 1.5,
+    });
+  };
+
+  return (
+    <>
```

Comment:
split compoenent,. and any com ponet shjould cross more than 50line

Reply:
Done